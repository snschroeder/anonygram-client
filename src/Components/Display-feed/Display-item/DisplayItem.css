.display-item {
    list-style-type: none;
    padding: 0;
    margin: 0;
}

.display-img {
    width: 100%;
    min-width: 288px;
    min-height: 288px;
    max-width: 500px;
    max-height: 500px;
    object-fit: cover;
}

.upvote-wrapper {
    position: relative;
    display: flex;
    direction: row;
    justify-content: center;
}

.upvote-count {
    position: absolute;
    bottom: 5px;
    right: 13px;
    font-size: 14px;
    font-weight: bold;
}

.upvote-button {
    position: absolute;
    z-index: auto;
    bottom: 10px;
    right: 30px;
    border: none;
    border-radius: 3px;
    width: 40px;
    height: 50px;
    background: grey;
    opacity: .5;
    color: white;
    font-size: 20px;
    font-weight: bold;
    text-align: center;
}

.upvote-button:hover,
.upvote-button:focus {
    opacity: 1;
    cursor: pointer;
}

.upvote-button-no-auth {
    position: absolute;
    z-index: auto;
    bottom: 10px;
    right: 30px;
    border: none;
    border-radius: 3px;
    width: 40px;
    height: 30px;
    background: grey;
    opacity: .5;
    color: white;
    font-size: 20px;
    font-weight: bold;
    text-align: center;
}

.delete-wrapper {
    position: relative;
    display: flex;
    direction: row;
    justify-content: center;
    align-items: center;
}

.delete-button {
    position: absolute;
    z-index: auto;
    bottom: 10px;
    right: 80px;
    border: none;
    border-radius: 3px;
    width: 40px;
    height: 30px;
    background: grey;
    opacity: .5;
    color: white;
    font-size: 20px;
    font-weight: bold;
    text-align: center;
}
.delete-icon {
    margin-top: 3px;
}

.upvote-button:hover,
.upvote-button:focus {
    opacity: 1;
    cursor: pointer;
}


@media screen and (min-width: 375px) {
    .display-img {
        min-width: 310px;
        min-height: 310px;
    }
  }
