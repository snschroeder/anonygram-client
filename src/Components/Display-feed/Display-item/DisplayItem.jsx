import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import './DisplayItem.css';
import { KeyboardArrowUp, DeleteOutline } from '@material-ui/icons';
import TokenService from '../../../services/token-service';

export default function DisplayItem(props) {
  const [loading, setLoading] = useState(true);
  const { imgAddress, upvotes, id, incrementUpvotes, handleDelete, userId } = props;
  const userIDFromToken = TokenService.parseAuthToken() !== undefined ? TokenService.parseAuthToken().id : ''; 
  const handleHasLoaded = () => setLoading(false)
  
  return (
    <li className="display-item">
      <Link to={`/p/${id}`}>
        <img className="display-img" src={imgAddress} alt="anonygram" onLoad={() => handleHasLoaded()}/>
      </Link>
      {!loading && userIDFromToken === userId ? <div className="delete-wrapper">
        <div className="delete-button">
          <DeleteOutline className="delete-icon" fontSize="small" onClick={() => handleDelete(id)} />
        </div>
      </div>
      :
      <></>
      }

      {!loading && userIDFromToken && userIDFromToken !== userId ? <div className="upvote-wrapper">
        <div className="upvote-button">
          <KeyboardArrowUp fontSize="large" onClick={() => incrementUpvotes(id)} />
          <p className="upvote-count">{upvotes}</p>
        </div>

      </div>
        :
        !loading && <div className="upvote-wrapper">
          <p className="upvote-button-no-auth">{upvotes}</p>
        </div>
      }
    </li>
  );
}
