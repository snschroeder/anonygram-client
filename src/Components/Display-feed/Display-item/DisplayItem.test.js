import React from 'react';
import ReactDOM from 'react-dom';
import DisplayItem from './DisplayItem';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import TokenService from '../../../services/token-service';

describe('DisplayItem component', () => {
  const props = {
    imgAddress:
      'https://lh3.googleusercontent.com/p/AF1QipMJcAXkTwnOj_TWYbMRwygy8ktducIF2kgOqSdW=s1600-w500',
    imgCaption: 'some image caption',
    upvotes: 12,
    id: 20,
    incrementUpvotes: () => {},
  };
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <DisplayItem
          imageAddress={
            'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRc53S16cGD_TTr80tPoxgIv5lsF3t52XjOuHV4Rognqp5qBdtO8A&s'
          }
          upvotes="20"
        />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders elements by default', () => {
    const wrapper = shallow(<DisplayItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders elements given props', () => {
    const props = {
      imgAddress:
        'https://lh3.googleusercontent.com/p/AF1QipMJcAXkTwnOj_TWYbMRwygy8ktducIF2kgOqSdW=s1600-w500',
      imgCaption: 'some image caption',
      upvotes: 12,
      id: 20,
      incrementUpvotes: () => {},
    };

    const wrapper = shallow(<DisplayItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  // setting state with Hooks is causing these to fail now
  //
  // it('renders upvote icon when user has auth token', () => {
  //   TokenService.saveAuthToken(
  //     'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImYxZTAyZjQyLTVhOTgtNDk4Yi05MzdiLTMyZDdiZGJjNDRkYSIsInVzZXJuYW1lIjoiYWRtaW5pc3RyYXRvciIsImlhdCI6MTU3NjI1NjMyMywiZXhwIjoxNTc2MjU5OTIzLCJzdWIiOiJhZG1pbmlzdHJhdG9yIn0.DbVBE8yMBGEUKqpqN3Pj4CKjWWAc708IiABn_8EVyqA'
  //   );
  //   const wrapper = shallow(<DisplayItem {...props} />);
  //   expect(toJson(wrapper)).toMatchSnapshot();
  //   TokenService.clearAuthToken();
  // });

  // it('initiates upvote call on click', () => {
  //   const mockIncrementUpvotes = jest.fn();
  //   TokenService.saveAuthToken(
  //     'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImYxZTAyZjQyLTVhOTgtNDk4Yi05MzdiLTMyZDdiZGJjNDRkYSIsInVzZXJuYW1lIjoiYWRtaW5pc3RyYXRvciIsImlhdCI6MTU3NjI1NjMyMywiZXhwIjoxNTc2MjU5OTIzLCJzdWIiOiJhZG1pbmlzdHJhdG9yIn0.DbVBE8yMBGEUKqpqN3Pj4CKjWWAc708IiABn_8EVyqA'
  //   );
  //   const wrapper = mount(
  //     <BrowserRouter><DisplayItem {...props} incrementUpvotes={mockIncrementUpvotes} /></BrowserRouter>
  //   );
  //   wrapper.find('KeyboardArrowUpIcon').simulate('click');
  //   expect(mockIncrementUpvotes).toHaveBeenCalled();
  // });
});
