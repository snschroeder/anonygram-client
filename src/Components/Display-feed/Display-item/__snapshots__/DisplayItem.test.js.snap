// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayItem component renders elements by default 1`] = `
<li
  className="display-item"
>
  <Link
    to="/p/undefined"
  >
    <img
      alt="anonygram"
      className="display-img"
      onLoad={[Function]}
    />
  </Link>
</li>
`;

exports[`DisplayItem component renders elements given props 1`] = `
<li
  className="display-item"
>
  <Link
    to="/p/20"
  >
    <img
      alt="anonygram"
      className="display-img"
      onLoad={[Function]}
      src="https://lh3.googleusercontent.com/p/AF1QipMJcAXkTwnOj_TWYbMRwygy8ktducIF2kgOqSdW=s1600-w500"
    />
  </Link>
</li>
`;
